src/torrent.cpp

```diff
@@ -4470,7 +4470,7 @@ namespace libtorrent {
 			avail_vec.push_back(i);
 		}
 
-		if (avail_vec.empty()) return piece_index_t(-1);
+		if (avail_vec.empty()) return {-1};
```

Comment:
I believe this only works if the implicit constructor is available, which is deprecated.

Reply:
true